resources/views/achievement/admin_includes/edit.blade.php

```diff
@@ -72,4 +108,20 @@
 
     </div>
 
-</form>
\ No newline at end of file
+</form>
+
+@section('javascript')
+    @parent
+    <script type="text/javascript">
+        let pageBlock = $('#achieve_page_block');
+        $('#has_page').on('click', function() {
+            if($(this).is(':checked')) {
+                pageBlock.show();
+                pageBlock.find('input').attr('required', true);
+            } else {
+                pageBlock.hide();
+                pageBlock.find('input').attr('required', false);
+            }
+        });
+    </script>
```

Comment:
If the state of checkbox is changed by other means than clicking it, this will not work

Reply:
I checked it and luckily it also works when changing the checkbox with the keyboard. https://stackoverflow.com/questions/11205957/jquery-difference-between-change-and-click-event-of-checkbox#answer-11251498